cpp/src/arrow/util/vector.h

```diff
@@ -31,8 +31,8 @@ namespace internal {
 
```

Comment:
We could also try to make this file internal IMHO.

Reply:
I'm holding back this one, as it is included in two examples:
https://github.com/apache/arrow/blob/005125790d83ecf50ecc20026d97021d7eb19fe2/cpp/examples/arrow/join_example.cc#L40
https://github.com/apache/arrow/blob/005125790d83ecf50ecc20026d97021d7eb19fe2/cpp/examples/arrow/execution_plan_documentation_examples.cc#L48

And I tend to not argue its internality (and its several transitive headers') in this poor PR. @pitrou What's your opinion? Thanks.

Reply:
Ok, let's keep it public then.